ui/flutter/lib/main.dart

```diff
@@ -67,23 +69,54 @@ Future<void> init(Args args) async {
   if (Util.isDesktop()) {
     await windowManager.ensureInitialized();
     final windowState = Database.instance.getWindowState();
+
+    // Check if menubar mode is enabled (only for macOS)
+    final runAsMenubarApp = Util.isMacos() && Database.instance.getRunAsMenubarApp();
+    
+    // Sync to SharedPreferences for native Swift to read on next startup
+    if (Util.isMacos()) {
+      Database.instance.saveRunAsMenubarApp(runAsMenubarApp);
+    }
+
     final windowOptions = WindowOptions(
       size: Size(windowState?.width ?? 800, windowState?.height ?? 600),
       center: true,
-      skipTaskbar: false,
+      skipTaskbar: runAsMenubarApp,
+      title: 'Gopeed',
     );
     await windowManager.waitUntilReadyToShow(windowOptions, () async {
       if (!args.hidden) {
         await windowManager.show();
         await windowManager.focus();
       }
+      // Apply menubar mode (hide dock icon) on macOS - must be after show()
```

Comment:
I have tested that the dock icon can be hidden through the skipTaskbar option in windowOptions above, so there is no need for this code here.